src/Helmet.js

```diff
@@ -552,8 +180,18 @@ const Helmet = (Component) => class HelmetWrapper extends React.Component {
             nestedChildren &&
             typeof nestedChildren !== "string"
         ) {
-            console.warn(`Helmet expects a single string as a child of ${child.type}`);
+            if (!VALID_TAG_NAMES.some(name => child.type === name)) {
```

Comment:
This is great checking, but only does VALID_TAG_NAMES check if the tag has children, can we maybe move that check higher to check all children while we are at it.